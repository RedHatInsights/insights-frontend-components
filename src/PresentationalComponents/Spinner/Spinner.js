import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './spinner.scss';

const Spinner = ({ centered, className, ...props }) => {

    let spinnerClasses = classNames(
        'ins-c-spinner',
        { [`ins-m-center`]: centered },
        className
    );

    return (
        <span className={ spinnerClasses } {...props} />
    );
};

Spinner.propTypes = {
    centered: PropTypes.bool,
    className: PropTypes.string
};


export default Spinner;
