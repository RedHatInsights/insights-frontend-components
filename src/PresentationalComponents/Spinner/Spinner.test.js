import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Spinner from './Spinner';

describe('TableToolbar component', () => {
    it('should render', () => {
        const wrapper = shallow(<Spinner/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render center Spinner', () => {
        const wrapper = shallow(<Spinner centered/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
