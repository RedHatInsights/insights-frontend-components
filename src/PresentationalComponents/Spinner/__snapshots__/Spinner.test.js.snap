// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableToolbar component should render 1`] = `
<span
  className="ins-c-spinner"
/>
`;

exports[`TableToolbar component should render center Spinner 1`] = `
<span
  className="ins-c-spinner ins-m-center"
/>
`;
